src/components/profile.ts

```diff
@@ -142,15 +151,20 @@ export const getUserProfileById = async (userId: string): Promise<UserProfile |
 
 // NOTE: data will always be just one customization, but we do not know which one, meaning many of
 // the loops are just to extract an unknown key-val pair
-export const editUserProfileById = async (userId: string, data: UserProfile): Promise<void> => {
+export const editUserProfileById = async (userId: string, data: UserProfile, message: Message): Promise<void> => {
```

Comment:
Also, I realized that you can replace the `message` parameter with the `member` obtained from this `message`. It looks like you needed the `message` parameter for the `member`, `userId`, and `guild`, but all of these can also be obtained from `message.member`. This would help with potential re-usability of helper functions.